boa_cli/src/debug/function.rs

```diff
@@ -0,0 +1,135 @@
+            direction => Err(JsNativeError::typ()
+                .with_message(format!("Unknown direction type '{direction}'"))
+                .into()),
+        };
+    }
+
+    Err(JsNativeError::typ()
+        .with_message("direction type must be a string")
+        .into())
+}
+
+/// Get functions instruction flowgraph
+fn flowgraph(_this: &JsValue, args: &[JsValue], context: &mut Context<'_>) -> JsResult<JsValue> {
+    let Some(value) = args.get(0) else {
+        return Err(JsNativeError::typ()
+        .with_message("expected function argument")
+        .into());
+    };
+
+    let Some(object) = value.as_object() else {
+        return Err(JsNativeError::typ()
+        .with_message(format!("expected object, got {}", value.type_of()))
+        .into());
+    };
+
+    let mut format = FlowgraphFormat::Mermaid;
+    let mut direction = Direction::LeftToRight;
+    if let Some(arguments) = args.get(1) {
+        if let Some(arguments) = arguments.as_object() {
+            format = flowgraph_parse_format_option(&arguments.get("format", context)?)?;
+            direction = flowgraph_parse_direction_option(&arguments.get("direction", context)?)?;
+        } else if value.is_string() {
+            format = flowgraph_parse_format_option(value)?;
+        } else {
+            return Err(JsNativeError::typ()
+                .with_message("options argument must be a string or object")
+                .into());
+        }
+    }
+
+    let object = object.borrow();
+
+    let Some(function) = object.as_function() else {
+        return Err(JsNativeError::typ()
+        .with_message("expected function object")
+        .into());
+    };
+
+    let code = match function {
+        Function::Ordinary { code, .. }
+        | Function::Async { code, .. }
+        | Function::Generator { code, .. }
+        | Function::AsyncGenerator { code, .. } => code,
+        Function::Native { .. } => {
+            return Err(JsNativeError::typ()
+                .with_message("native functions do not have bytecode")
+                .into())
+        }
+    };
+
+    let mut graph = Graph::new(direction);
+    code.to_graph(context.interner(), graph.subgraph(String::default()));
+    let result = match format {
+        FlowgraphFormat::Graphviz => graph.to_graphviz_format(),
+        FlowgraphFormat::Mermaid => graph.to_mermaid_format(),
+    };
+
+    Ok(JsValue::new(result))
+}
+
+/// Trace function.
+fn trace(_: &JsValue, args: &[JsValue], context: &mut Context<'_>) -> JsResult<JsValue> {
+    let value = args.get_or_undefined(0);
+    let this = args.get_or_undefined(1);
+
+    let Some(callable) = value.as_callable() else {
+        return Err(JsNativeError::typ()
+        .with_message("expected callable object")
+        .into());
+    };
+
+    let arguments = args.get(2..).unwrap_or(&[]);
+
+    context.set_trace(true);
+    let result = callable.call(this, arguments, context);
+    context.set_trace(false);
```

Comment:
It would be good to have a `trace` property on the functions itself, since this'll also trace called functions which may pollute the output.